skycatalogs/utils/config_utils.py

```diff
@@ -16,12 +16,81 @@
 CURRENT_SCHEMA_VERSION = '1.2.0'
 
 
+class YamlIncludeLoader(yaml.SafeLoader):
+
+    """YAML Loader that supports file include directives.
+
+    Uses ``!include`` directive in a YAML file to point to another
+    YAML file to be included. The path in the include directive is relative
+    to the top-level file
+
+        storageClasses: !include storageClasses.yaml
+
+    Examples
+    --------
+    >>> with open("document.yaml", "r") as f:
+           data = yaml.load(f, Loader=YamlIncludeLoader)
+
+    Parameters
+    ----------
+    stream :  text io stream
+        The stream to parse.
+
+    This code was adapted from the LSST Science Pipelines Butler.
+    """
+    def __init__(self, filestream):
+        super().__init__(filestream)
+        self._root = os.path.dirname(filestream.name)
+        self.add_constructor("!include", YamlIncludeLoader.include)
+        self._current_dir = self._root
+
+    def include(self, node: yaml.Node) -> list[Any] | dict[str, Any]:
+        result: list[Any] | dict[str, Any]
+        if isinstance(node, yaml.ScalarNode):
+            return self.extractFile(self.construct_scalar(node))  # type: ignore[arg-type]
+
+        elif isinstance(node, yaml.SequenceNode):
+            result = []
+            for filename in self.construct_sequence(node):
+                result.append(self.extractFile(filename))
+            return result
+
+        elif isinstance(node, yaml.MappingNode):
+            result = {}
+            for k, v in self.construct_mapping(node).items():
+                if not isinstance(k, str):
+                    raise TypeError(f"Expected only strings in YAML mapping; got {k!r} of type {type(k)}.")
+                result[k] = self.extractFile(v)
+            return result
+
+        else:
+            print("Error:: unrecognised node type in !include statement",
+                  file=sys.stderr)
+            raise yaml.constructor.ConstructorError
+
+    def extractFile(self, filepath: str) -> Any:
+        if filepath.startswith('/'):
+            actual_path = filepath
+        else:
+            actual_path = os.path.join(self._current_dir, filepath)
+        print("Opening YAML file via !include: %s", actual_path)
+
+        old_current = self._current_dir
+        self._current_dir = actual_path
+        # Read all the data from the resource
+        with open(actual_path) as f:
+            content = yaml.load(f, YamlIncludeLoader)
+        self._current_dir = old_current
```

Comment:
You don't need to do any special handling of `self._current_dir` here since it's set per-instance of `YamlIncludeLoader`.  So this code can just be
```
# Read all the data from the resource
with open(actual_path) as f:
    content = yaml.load(f, YamlIncludeLoader)
```
The other lines don't have any net effect, otherwise it would have been necessary to set `self._current_dir = os.path.dirname(actual_path)` on line 79.